graphql/resolvers.go

```diff
@@ -1819,6 +1819,76 @@ func (r *mutationResolver) UnschedulePatchTasks(ctx context.Context, patchID str
 	return &patchID, nil
 }
 
+// ScheduleUndispatchedBaseTasks only allows scheduling undispatched base tasks for tasks that are failing on the current patch
+func (r *mutationResolver) ScheduleUndispatchedBaseTasks(ctx context.Context, patchID string) ([]*restModel.APITask, error) {
+	opts := data.TaskFilterOptions{
+		Statuses:              evergreen.TaskFailureStatuses,
+		IncludeExecutionTasks: true,
+	}
+	tasks, _, err := r.sc.FindTasksByVersion(patchID, opts)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch: %s ", err.Error()))
+	}
+
+	restartedTasks := []*restModel.APITask{}
+	generatedTasksToRestart := []task.Task{}
+
+	for _, t := range tasks {
+		// If a task is a generator task or generated don't schedule it until we get all of the generated tasks we want to generate
+		if !t.GenerateTask && t.GeneratedBy == "" {
+			// We can ignore an error while fetching tasks because this could just mean the task didn't exist on the base commit.
+			baseTask, _ := t.FindTaskOnBaseCommit()
+			if baseTask != nil {
+				task, err := SetScheduled(ctx, r.sc, baseTask.Id, true)
+				if err != nil {
+					return nil, err
+				}
+				restartedTasks = append(restartedTasks, task)
+			}
+			// If a task is generated lets keep track of it so we can restart it later
+		} else if t.GeneratedBy != "" {
+			generatedTasksToRestart = append(generatedTasksToRestart, t)
+
+		}
+	}
+
+	baseGeneratorTasksToSchedule := make(map[string]*task.Task)
+	// If a generated task does not have a base task find its generator and use its base task
+	// to generate it
+	for _, t := range generatedTasksToRestart {
+		baseTask, _ := t.FindTaskOnBaseCommit()
+		// If the task is undispatched or doesn't exist on the base commit then we want to schedule
+		if baseTask == nil || baseTask.Status == evergreen.TaskUndispatched {
+			generatorTask, err := task.FindByIdExecution(t.GeneratedBy, nil)
+			if err != nil {
+				return nil, InternalServerError.Send(ctx, fmt.Sprintf("Experienced an error trying to find the generator task: %s", err.Error()))
+			}
+			if generatorTask != nil {
+				baseGeneratorTask, _ := generatorTask.FindTaskOnBaseCommit()
+				// If baseGeneratorTask is nil then it didn't exist on the base task and we can't do anything
+				if baseGeneratorTask != nil {
+					err = baseGeneratorTask.SetGeneratedTasksToStepback(t.BuildVariant, t.DisplayName)
```

Comment:
Sorry for the app thought, but Stepback is now the wrong term, because this operation isn't stepback (we talked about this briefly in the chat). I would change this to SetGeneratedTasksToActivate (you can also change the bson tags for the variable itself since it isn't used much yet).